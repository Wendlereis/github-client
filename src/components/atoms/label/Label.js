import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

const StyledLabel = styled.label`
  font-family: menlo;
  font-size: 12px;
  color: rgba(0, 0, 0, 0.54);
  text-transform: uppercase;

  ${props => {
    return (
      props.title &&
      css`
        font-size: 44px;
      `
    )
  }}

  ${props => {
    return (
      props.subtitle &&
      css`
        font-size: 20px;
        text-transform: none;
      `
    )
  }}
`

export default function Label({ text, ...rest }) {
  return <StyledLabel {...rest}>{text}</StyledLabel>
}

Label.propTypes = {
  text: PropTypes.string.isRequired
}
