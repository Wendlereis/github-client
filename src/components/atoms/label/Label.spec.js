import React from 'react'
import { shallow } from 'enzyme'
import Label from './Label'

describe('Label', () => {
  it('should render correctly with a text', () => {
    const component = shallow(<Label text="myText" />)

    expect(component).toMatchSnapshot()
  })

  it('should render correctly with a text as a title', () => {
    const component = shallow(<Label title text="myText" />)

    expect(component).toMatchSnapshot()
  })

  it('should show repository name as a title', () => {
    const repoName = 'My Repository'
    const component = shallow(<Label title text={repoName} />)

    expect(component).toMatchSnapshot()
  })
})
