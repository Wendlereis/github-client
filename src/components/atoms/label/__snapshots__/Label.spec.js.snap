// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render correctly with a text 1`] = `
<styled.label>
  myText
</styled.label>
`;

exports[`Label should render correctly with a text as a title 1`] = `
<styled.label
  title={true}
>
  myText
</styled.label>
`;

exports[`Label should show repository name as a title 1`] = `
<styled.label
  title={true}
>
  My Repository
</styled.label>
`;
